docs/cache-server.md

```diff
@@ -0,0 +1,125 @@
+Thus, the cache server serves as a central place for shards to store and read common data.
+
+### High-level overview
+
+The cache server is a regular, Kubernetes-style CRUD API server with support of LIST/WATCH semantics. 
+Conceptually it supports two modes of operations. 
+The first mode is in which a shard gets its private space for storing its own data. 
+The second mode is in which a shard can read other shards' data.
+
+The first mode of operation is implemented as a controller that runs within a shard. 
+It maintains an informer for `LIST/WATCH`ing its own resources. 
+It compares received data to its local copy and makes sure both copies are consistent.
+
+The second mode of operation is implemented as a separate informer within a shard. 
+Thanks to that a shard can implement a different resiliency strategy. 
+For example, it can tolerate the unavailability of the secondary during startup and become ready.
+
+### Running the server
+
+The cache server can be run as a standalone binary or as part of a kcp server.
+
+The standalone binary is in https://github.com/kcp-dev/kcp/tree/main/cmd/cache-server and can be run by issuing `go run ./cmd/cache-server/main.go` command.
+
+To run it as part of a kcp server, pass `--cache-url` flag to the kcp binary.
+
+### Client-side functionality
+
+In order to interact with the cache server from a shard, the https://github.com/kcp-dev/kcp/tree/main/pkg/cache/client 
+repository provides the following client-side functionality:
+
+[ShardRoundTripper](https://github.com/kcp-dev/kcp/blob/b739fa5b5c83fb2c43b631c6234264d5dd1fc6e4/pkg/cache/client/round_tripper.go#L54), 
+a shard aware wrapper around `http.RoundTripper`. It changes the URL path to target a shard from the context.
+
+[DefaultShardRoundTripper](https://github.com/kcp-dev/kcp/blob/b739fa5b5c83fb2c43b631c6234264d5dd1fc6e4/pkg/cache/client/round_tripper.go#L128) 
+is a `http.RoundTripper` that sets a default shard name if not specified in the context.
+
+For example, in order to make a client shard aware, inject the `http.RoundTrippers` to a `rest.Config`
+
+```
+import(
+cacheclient "github.com/kcp-dev/kcp/pkg/cache/client"
+"github.com/kcp-dev/kcp/pkg/cache/client/shard"
+)
+
+// adds shards awareness to a client with a default `shard.Wildcard` name.
+NewClusterForConfig(cacheclient.WithShardRoundTripper(cacheclient.WithDefaultShardRoundTripper(serverConfig.LoopbackClientConfig, shard.Wildcard)))
+
+// and then change the context when you need to access a specific shard and pass is when making a HTTP request
+ctx = cacheclient.WithShardInContext(ctx, shard.New("cache"))
+```
+
+### Authorization/Authentication
+
+Not implemented at the moment
+
+### Build-in resources
+
+Out of the box, the server supports the following resources:
+- `apiresourceschemas`
+- `apiexports`
+
+All those resources are stored in `system:cache:server` shard under `system:system-crds` cluster.
+
+### Adding new resources
```

Comment:
Do we expect the built-in ones to remain built-in forever? Could we instead have a section here on "how do you add data to the cache server?" and lay out the near-term (hard-code) and long-term (replication claim) visions?

Reply:
> Do we expect the built-in ones to remain built-in forever?

I think so, some resource like `apiresourceschemas`, `apiexports` or `secrets`, `configmaps` will always be required/availiable